tests/crontime.test.ts

```diff
@@ -451,7 +451,18 @@ describe('crontime', () => {
 			currentDate = nextDate;
 		}
 	});
-	it('Should schedule jobs inside time zone changes that shifts time forward to the end of the shift, for weekly jobs', () => {
+	it('should not execute immediately if conditions have not been met during forward DST jump', () => {
+		// Europe/Paris DST starts at 30 Mar 2025, 02:00 (+1 to hours)
+		let currentDate = DateTime.fromISO('2025-03-30T01:59', {
+			zone: 'Europe/Paris'
+		});
+		const cronTime = new CronTime('20 4 * * *');
+		const nextDate = cronTime.getNextDateFrom(currentDate, 'Europe/Paris');
+		expect(nextDate.toString()).toEqual(
+			DateTime.fromISO('2025-03-30T04:20', { zone: 'Europe/Paris' }).toString()
+		);
+	});
```

Comment:
While reviewing the PR, I had a hunch that because we are not verifying whether an execution was missed and returning the correct execution time like the previous `_forwardDSTJump` did, we would always end up executing when a jump happens, even though no execution is planned during the jump.

I've added a test case to verify this thought, and this is indeed happening:

![image](https://github.com/user-attachments/assets/27f1683c-f23f-4123-8a0f-5ffb11e74a27)

Reply:
I have cherry-picked it on the main branch for confirmation, and it doesn't happen there.

Reply:
yeah I think I might have been too aggressive removing that function. I was just so excited to detect a jump by checking if `offset` changed. I might add it back or a similar helper function that still uses offset to track down the exact time that the change happens

Reply:
fixed so your test case passes now! thanks for adding that